cpp/input/event.h

```diff
@@ -0,0 +1,149 @@
+// Copyright 2015-2015 the openage authors. See copying.md for legal info.
+
+#ifndef OPENAGE_INPUT_EVENT_H_
+#define OPENAGE_INPUT_EVENT_H_
+
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+#include <SDL2/SDL.h>
+
+namespace openage {
+namespace input {
+
+/**
+ * highest level classes of input
+ */
+enum class event_class {
+	ANY,
+	KEYBOARD,
+	TEXT,
+	ALPHA,			// abc
+	DIGIT,			// 123
+	PRINT,			// remaining printable chars
+	NONPRINT,		// tab, return, backspace, delete
+	OTHER,			// arrows, home, end
+	MOUSE,
+	MOUSE_BUTTON,
+	MOUSE_WHEEL,
+	MOUSE_MOTION
+};
+
+
+struct event_class_hash {
+	int operator()(const event_class &s) const;
+};
+
+
+/**
+ * each event type mapped to parent type
+ */
+static std::unordered_map<event_class, event_class, event_class_hash> event_base {
+	{event_class::KEYBOARD, event_class::ANY},
+	{event_class::TEXT, event_class::KEYBOARD},
+	{event_class::ALPHA, event_class::TEXT},
+	{event_class::DIGIT, event_class::TEXT},
+	{event_class::PRINT, event_class::TEXT},
+	{event_class::NONPRINT, event_class::KEYBOARD},
+	{event_class::OTHER, event_class::KEYBOARD},
+	{event_class::MOUSE, event_class::ANY},
+	{event_class::MOUSE_BUTTON, event_class::MOUSE},
+	{event_class::MOUSE_WHEEL, event_class::MOUSE},
+	{event_class::MOUSE_MOTION, event_class::MOUSE},
+};
+
+/**
+ * mods set on an event
+ */
+enum class modifier {
+	CTRL,
+	ALT,
+	SHIFT
+};
+
+
+struct modifier_hash {
+	int operator()(const modifier &s) const;
+};
+
+
+/**
+ * types used by events
+ */
+using code_t = int;
+using modset_t = std::unordered_set<modifier, modifier_hash>;
+
+
+/**
+ * a set sdl event types
+ * TODO to char function
+ */
+struct class_code_t {
+public:
+	class_code_t(event_class cl, code_t code);
```

Comment:
with constructor and member functions, this is most likely a class, and especially not a `_t` type right?